File: humanize.go
// Package humanize provides tiny zero-dependency helpers to humanize values:
// byte sizes, durations, English ordinals, and pluralization.
package humanize

import (
	"strconv"
	"strings"
)

var byteUnits = []string{"KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB"}

func trim(n float64) string {
	s := strconv.FormatFloat(n, 'f', 1, 64)
	return strings.TrimSuffix(s, ".0")
}

// FormatBytes formats a byte count as a human-readable string using binary
// (1024) units, e.g. FormatBytes(1536) == "1.5 KB".
func FormatBytes(bytes int64) string {
	n := float64(bytes)
	if n < 1024 {
		return strconv.FormatInt(bytes, 10) + " B"
	}
	last := len(byteUnits) - 1
	for i, unit := range byteUnits {
		n /= 1024
		if n < 1024 || i == last {
			return trim(n) + " " + unit
		}
	}
	return trim(n) + " YB"
}

// FormatDuration formats a duration in seconds as a human-readable string,
// e.g. FormatDuration(3661) == "1h 1m 1s".
func FormatDuration(seconds int64) string {
	if seconds == 0 {
		return "0s"
	}
	var parts []string
	days := seconds / 86400
	seconds %= 86400
	hours := seconds / 3600
	seconds %= 3600
	minutes := seconds / 60
	seconds %= 60
	if days > 0 {
		parts = append(parts, strconv.FormatInt(days, 10)+"d")
	}
	if hours > 0 {
		parts = append(parts, strconv.FormatInt(hours, 10)+"h")
	}
	if minutes > 0 {
		parts = append(parts, strconv.FormatInt(minutes, 10)+"m")
	}
	if seconds > 0 {
		parts = append(parts, strconv.FormatInt(seconds, 10)+"s")
	}
	return strings.Join(parts, " ")
}

// Ordinal returns n with its English ordinal suffix, e.g. Ordinal(21) == "21st".
func Ordinal(n int) string {
	suffix := "th"
	if m := n % 100; m < 11 || m > 13 {
		switch n % 10 {
		case 1:
			suffix = "st"
		case 2:
			suffix = "nd"
		case 3:
			suffix = "rd"
		}
	}
	return strconv.Itoa(n) + suffix
}

// Pluralize returns "<count> <singular|plural>", choosing the form based on
// count, e.g. Pluralize(3, "file", "files") == "3 files".
func Pluralize(count int, singular, plural string) string {
	word := plural
	if count == 1 {
		word = singular
	}
	return strconv.Itoa(count) + " " + word
}

File: Humanize.php
<?php

declare(strict_types=1);

namespace Garudadidada;

/**
 * Tiny zero-dependency helpers to humanize values:
 * byte sizes, durations, English ordinals, and pluralization.
 */
final class Humanize
{
    /** @var string[] */
    private const BYTE_UNITS = ['KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];

    /**
     * Format a byte count as a human-readable string using binary (1024) units.
     * formatBytes(1536) === "1.5 KB"
     */
    public static function formatBytes(int $bytes): string
    {
        $n = (float) $bytes;
        if ($n < 1024) {
            return $bytes . ' B';
        }
        $last = count(self::BYTE_UNITS) - 1;
        foreach (self::BYTE_UNITS as $i => $unit) {
            $n /= 1024;
            if ($n < 1024 || $i === $last) {
                return self::trim($n) . ' ' . $unit;
            }
        }
        return self::trim($n) . ' YB';
    }

    /**
     * Format a duration in seconds as a human-readable string.
     * formatDuration(3661) === "1h 1m 1s"
     */
    public static function formatDuration(int $seconds): string
    {
        if ($seconds === 0) {
            return '0s';
        }
        $parts = [];
        $days = intdiv($seconds, 86400);
        $seconds %= 86400;
        $hours = intdiv($seconds, 3600);
        $seconds %= 3600;
        $minutes = intdiv($seconds, 60);
        $seconds %= 60;
        if ($days > 0) {
            $parts[] = $days . 'd';
        }
        if ($hours > 0) {
            $parts[] = $hours . 'h';
        }
        if ($minutes > 0) {
            $parts[] = $minutes . 'm';
        }
        if ($seconds > 0) {
            $parts[] = $seconds . 's';
        }
        return implode(' ', $parts);
    }

    /**
     * Return n with its English ordinal suffix.
     * ordinal(21) === "21st"
     */
    public static function ordinal(int $n): string
    {
        $suffix = 'th';
        $m = $n % 100;
        if ($m < 11 || $m > 13) {
            switch ($n % 10) {
                case 1:
                    $suffix = 'st';
                    break;
                case 2:
                    $suffix = 'nd';
                    break;
                case 3:
                    $suffix = 'rd';
                    break;
            }
        }
        return $n . $suffix;
    }

    /**
     * Return "<count> <singular|plural>" choosing the form based on count.
     * pluralize(3, "file") === "3 files"
     */
    public static function pluralize(int $count, string $singular, ?string $plural = null): string
    {
        $plural = $plural ?? $singular . 's';
        $word = $count === 1 ? $singular : $plural;
        return $count . ' ' . $word;
    }

    private static function trim(float $n): string
    {
        $s = number_format($n, 1, '.', '');
        if (substr($s, -2) === '.0') {
            $s = substr($s, 0, -2);
        }
        return $s;
    }
}
